### PR TITLE
feature: display more cards behind the first one

### PR DESCRIPTION
**Pull request by giboin**
## Description

If you want to use widgets in your cards that don't build instantly, like network images or videos, the cards might take time to appear and if the users swipe fast enough they might experience some lag.
I wrote this code that allows you to build more cards at a time and display them behind the front one, in order to help optimize perfomance and make navigation more fluid.

**EDIT**
I got tired of fighting with git rebase so I just created a clean new fork and applied my changes, closed the old PR and opened this new one

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/ricardodalarme/flutter_card_swiper/blob/main/CONTRIBUTING.md
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/ricardodalarme/flutter_card_swiper/blob/chore/add-a-contributing-guide/CONTRIBUTING.md#version
[following repository CHANGELOG style]:https://github.com/ricardodalarme/flutter_card_swiper/blob/chore/add-a-contributing-guide/CONTRIBUTING.md#changelog

## Visual reference

without the feature:

https://user-images.githubusercontent.com/81298073/220890429-3a363c15-fb0d-4753-8e21-40b0c5c9dc4a.mp4

with the feature (numberOfCardsDisplayed =5):

https://user-images.githubusercontent.com/81298073/220890569-103f4106-3965-4477-947c-0b5e76f06bee.mp4

### CODE REVIEW

**Comment by giboin** [comment]
As I say in the EDIT part of my description this is a new PR, exactly like #5 but with only one commit (didn't need more for such a little change) and no conflicts with the base branch. 
I don't understand why checks keep failing though.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## [2.0.2]
```
Comment (ricardodalarme) [inline CHANGELOG.md:1]: As this is a brand new feature, it should be:

```suggestion
## [2.1.0]
```

File: lib/src/card_swiper.dart [hunk 2]
```diff
@@ -149,13 +157,27 @@ class _CardSwiperState<T extends Widget> extends State<CardSwiper<T>>
...
             builder: (BuildContext context, BoxConstraints constraints) {
+              //the number of cards that are built on the screen
+              final stackSize = widget.isLoop
+                  ? widget.numberOfCardsDisplayed
+                  : _stack.isNotEmpty
+                      ? min(
+                          widget.numberOfCardsDisplayed,
+                          _stack.length,
+                        )
+                      : 0;
```
Comment (ricardodalarme) [inline lib/src/card_swiper.dart:168]: Can you move this to a function?

File: lib/src/card_swiper.dart [hunk 3]
```diff
@@ -149,13 +157,27 @@ class _CardSwiperState<T extends Widget> extends State<CardSwiper<T>>
...
-                  if (_stack.isNotEmpty) _frontItem(constraints),
-                ],
+                children: List.generate(stackSize, (index) {
+                  if (index == 0) {
+                    return _frontItem(constraints);
+                  } else if (index == 1) {
+                    return _secondItem(constraints);
+                  } else {
+                    return _backItem(constraints, index);
+                  }
```
Comment (ricardodalarme) [inline lib/src/card_swiper.dart:179]:
```suggestion
                  if (index == 0) {
                    return _frontItem(constraints);
                  }
                  if (index == 1) {
                    return _secondItem(constraints);
                  }
                  return _backItem(constraints, index);
```

File: lib/src/card_swiper.dart [hunk 4]
```diff
@@ -149,13 +157,27 @@ class _CardSwiperState<T extends Widget> extends State<CardSwiper<T>>
...
+                      ? min(
+                          widget.numberOfCardsDisplayed,
+                          _stack.length,
+                        )
+                      : 0;
               return Stack(
                 clipBehavior: Clip.none,
                 fit: StackFit.expand,
-                children: [
-                  if (_hasBackItem) _backItem(constraints),
```
Comment (ricardodalarme) [inline lib/src/card_swiper.dart:156]: Since you are no longer using the _hasBackItem getter, you can remove it

**Comment by ricardodalarme** [comment]
Your PR is really good, it will be a good addition. Thank you very much! I just left some comments, can you check them, please?

**Comment by ricardodalarme** [comment]
And don't worry about the CI, it breaks because it can't report tests from a fork. I'll fix that later.

**Comment by giboin** [comment]
Thank you for your review, I did everything you suggested!

**Review by ricardodalarme — COMMENTED** [review]
Nice one!

**Review by ricardodalarme — APPROVED** [review]

**Pull request merged**